aioamqp/channel.py

```diff
@@ -778,6 +780,32 @@ def basic_recover_ok(self, frame):
         future.set_result(True)
         logger.debug("Cancel ok")
 
+    @asyncio.coroutine
+    def basic_return(self, frame):
+        response = amqp_frame.AmqpDecoder(frame.payload)
+        reply_code = response.read_short()
+        reply_text = response.read_shortstr()
+        exchange_name = response.read_shortstr()
+        routing_key = response.read_shortstr()
+        content_header_frame = yield from self.protocol.get_frame()
+
+        buffer = io.BytesIO()
+        while buffer.tell() < content_header_frame.body_size:
+            content_body_frame = yield from self.protocol.get_frame()
+            buffer.write(content_body_frame.payload)
+
+        body = buffer.getvalue()
+        envelope = ReturnEnvelope(reply_code, reply_text,
+                                  exchange_name, routing_key)
+        properties = content_header_frame.properties
+        callback = self.return_callback
```

Comment:
This should be added as an optional keyword parameter to the `queue_declare` coroutine

Reply:
Should it be part of queue declare? I am not so sure. A queue isnt related to a retun at all. You have get a return packet without ever declaring a queue. Not sure they are related.

Reply:
Oh, got it, I just checked the doc.

To be honest, I think it should be declared at channel creation time.
Imho, an addition to the `Channel`'s `__init__` function with a default to None should work shouldn't it ?

So, each time we publish a message through a channel, we could set the function when creating it, to receive failed messages.